pykafka/simpleconsumer.py

```diff
@@ -462,6 +477,15 @@ def _handle_success(parts):
                     [part for errcode, parts in parts_by_error.iteritems()
                      for part in parts])
 
+            # release all locks to allow fetching
+            for errcode, owned_partitions in parts_by_error.iteritems():
```

Comment:
So if we get here, and we're at our `self._offsets_reset_max_retries`th retry, I think this may leave the errored partitions at some other offset than specified - could we raise an exception at that point?